oak-run-commons/src/main/java/org/apache/jackrabbit/oak/index/IndexerSupport.java

```diff
@@ -230,4 +233,15 @@ public Set<String> getPreferredPathElements(Set<IndexDefinition> indexDefinition
     public <T> Predicate<T> getFilterPredicate(Set<IndexDefinition> indexDefinitions, Function<T, String> typeToRepositoryPath) {
         return t -> indexDefinitions.stream().anyMatch(indexDef -> indexDef.getPathFilter().filter(typeToRepositoryPath.apply(t)) != PathFilter.Result.EXCLUDE);
     }
+
+    /**
+     *
+     * @param pattern Pattern for a custom excludes regex based on which paths would be filtered out
+     * @param typeToRepositoryPath Function to convert type <T> to valid repository path of type <String>
+     * @param <T>
+     * @return filter predicate based on a matcher for a custom excludes regex based on which paths would be filtered out
+     */
+    public <T> Predicate<T> getFilterPredicateBasedOnCustomRegex(Pattern pattern, Function<T, String> typeToRepositoryPath) {
```

Comment:
Could you write a comment about what this function is supposed to do? After looking at it shortly, I'm still not 100 % sure what problem it solves and why. Would also like an example in the Javadocs.

Reply:
changed the language a bit to make it more clear.